src/test/java/api/support/fakes/FakeStorageModule.java

```diff
@@ -128,15 +128,17 @@ private void create(RoutingContext routingContext) {
 
       body.put("id", id);
 
-      if(includeChangeMetadata) {
+    if(includeChangeMetadata) {
         final String fakeUserId = APITestContext.getUserId();
+        final String fakePatronGroupId = APITestContext.getPatronGroupIdAtCheckoutId();
         body.put(changeMetadataPropertyName, new JsonObject()
           .put("createdDate", new DateTime(DateTimeZone.UTC)
             .toString(ISODateTimeFormat.dateTime()))
           .put("createdByUserId", fakeUserId)
           .put("updatedDate", new DateTime(DateTimeZone.UTC)
             .toString(ISODateTimeFormat.dateTime()))
-          .put("updatedByUserId", fakeUserId));
+          .put("updatedByUserId", fakeUserId))
+          .put("patronGroupIdAtCheckout", fakePatronGroupId);;
```

Comment:
This fake is intended to represent the behaviour of a storage module.

The real storage module will not be able to add in this information, which likely invalidates most of the tests which rely on this behaviour.

Can you help me understand the intent behind changing the fake to put this information on all records?

Reply:
The storage has already been updated with the new field folio-org/mod-circulation-storage#177

Reply:
@mrdpopov I’m aware that the storage module has had a new property added.

How does that relate to my question?

The fake storage module does not need changing for each property introduced.

The storage module cannot / does not set that new property itself, yet this is what this line in the fake does.

Can you help me understand what this change is intended to achieve?

Reply:
@marcjohnson-kint So, I removed these changes in the FakeStorageModule. 

I added this field ("patronGroupIdAtCheckout") to the builder for loan setup.

Reply:
@VRohach @mrdpopov Can you help me understand why this field is being set by the client (in this case the tests, by using the builder)?

My understanding of the intent of this change is to set this property during the check out process. Have I missed a use case where there is a need for a client to specify it?

Reply:
@VRohach Thank you for removing this change from the fake storage module.

And thank you for helping me understand that there needs to be better documentation around the testing approach and tooling used in this module